src/pages/CommandView/form-data/get-submit-argument/prepare-model-for-submit.ts

```diff
@@ -0,0 +1,299 @@
+      return null
+    }
+
+    throw new Error(
+      `Received parameter model without needed key '${parameterKey}'`,
+    )
+  } else {
+    if (parameter.optional) {
+      if (typeof modelParams[parameterKey] === 'undefined') {
+        return null
+      }
+    } else if (!parameter.multi) {
+      if (typeof modelParams[parameterKey] === 'undefined') {
+        throw new Error(
+          `Received empty parameter value for non-optional key '${key}'`,
+        )
+      }
+    }
+  }
+
+  const parameterValue = modelParams[parameterKey]
+
+  const isAny = parameter.type === 'Any'
+  const isMulti = parameter.multi
+  const isAnyList = isAny && isMulti
+  const hasSubparameters = parameter.parameters.length > 0
+
+  if (isAny) {
+    if (parameterValue === '') {
+      return { [parameterKey]: null }
+    }
+    if (isAnyList) {
+      if (!Array.isArray(parameterValue)) {
+        throw new Error(`Model has non-array value for key '${parameterKey}'`)
+      }
+
+      const parameterValues = parameterValue as string[]
+      const fixedSubParametersArray = parameterValues.map(parseAny)
+
+      return { [parameterKey]: fixedSubParametersArray }
+    } else {
+      return { [parameterKey]: parseAny(parameterValue as string) }
+    }
+  } else if (hasSubparameters) {
+    if (isMulti) {
+      if (!Array.isArray(parameterValue)) {
+        throw new Error(`Received non-array value for key '${parameterKey}'`)
+      }
+
+      let fixedSubParameters = {} as ObjectWithStringKeys
+      const parameterValues = parameterValue as ObjectWithStringKeys[]
+      const fixedSubParametersArray = [] as ObjectWithStringKeys[]
+
+      for (const value of parameterValues) {
+        for (const subParameter of parameter.parameters) {
+          const subParameterKey = key + '.' + subParameter.key
+          const fixedSubParameter = fixAny(value, subParameter, subParameterKey)
+
+          fixedSubParameters = {
+            ...fixedSubParameters,
+            ...fixedSubParameter,
+          }
+        }
+        fixedSubParametersArray.push(fixedSubParameters)
+        fixedSubParameters = {}
+      }
+
+      return { [parameterKey]: fixedSubParametersArray }
+    } else {
+      let fixedSubParameters = {} as ObjectWithStringKeys
+
+      for (const subParameter of parameter.parameters) {
+        const subParameterKey = key + '.' + subParameter.key
+        const fixedSubParameter = fixAny(
+          parameterValue as ObjectWithStringKeys,
+          subParameter,
+          subParameterKey,
+        )
+
+        fixedSubParameters = {
+          ...fixedSubParameters,
+          ...fixedSubParameter,
+        }
+      }
+      fixedSubParameters = {
+        ...(parameterValue as ObjectWithStringKeys),
+        ...fixedSubParameters,
+      }
+
+      const returnValue = { [parameterKey]: fixedSubParameters }
+
+      return returnValue
+    }
+  }
+
+  return { [parameterKey]: parameterValue }
+}
+
+const prepareModelForSubmit = (
```

Comment:
This does some fixup of the submit payload for the `Dictionary` and `Any` parameter types, which are particularly troublesome.